core/src/main/java/com/linecorp/armeria/client/ClientFactoryOptions.java

```diff
@@ -72,7 +72,8 @@ public final class ClientFactoryOptions
     public static final ClientFactoryOption<Function<? super EventLoopGroup, ? extends EventLoopScheduler>>
             EVENT_LOOP_SCHEDULER_FACTORY = ClientFactoryOption.define(
             "EVENT_LOOP_SCHEDULER_FACTORY",
-            eventLoopGroup -> new DefaultEventLoopScheduler(eventLoopGroup, 0, 0, ImmutableList.of()));
+            eventLoopGroup -> new DefaultEventLoopScheduler(eventLoopGroup, 0, 0, ImmutableList.of(),
+                                                            Flags.defaultClientIdleTimeoutMillis()));
```

Comment:
Can we somehow make the factory get the configured idle timeout from the current `ClientFactoryOptions`? Maybe defining a new factory interface that replaces `Function`?

Reply:
Sounds like a good idea.

Reply:
This PR is targeted for the patch version. May I add the new factory interface in the next minor version? I'm a little hesitant about adding a new API to the patch version.

Reply:
Created. https://github.com/line/armeria/issues/5289
If you still think it is good to add the API in this PR, please let me know.

Reply:
Just let me know when this is ready. 😉